database/repositories/image_repository.py

```diff
@@ -0,0 +1,87 @@
+from database.tables.image import *
+
+
+# This class is responsible for doing CRUD operations on the Image Table in DB
+# This class was written as per the Repository Model which allows us to change the DB in the future without changing
+# business code
+class ImageRepository:
+    def __init__(self):
+        self.connection = None
+
+    # inserts the arguments into table
+    # Commits if inserted successfully otherwise rollbacks if some issue occured and bubbles the exception
+    def insert(self, image_name, project_id, is_public=False):
+        try:
+            self.connection = DatabaseConnection()
+            img = Image()
+            img.name = image_name
+            img.project_id = project_id
+            img.is_public = is_public
+            self.connection.session.add(img)
+            self.connection.session.commit()
+        # should change to more specific exception
+        except Exception:
+            self.connection.session.rollback()
+            raise
+        finally:
+            self.connection.close()
+
+    # deletes images with name under the given project name
+    # commits if deletion was successful otherwise rollback occurs and exception is bubbled up
+    def delete_with_name_from_project(self, name, project_name):
```

Comment:
I feel like all those methods should have a decorator, and handle the exception catching and rollback there.